### DISCUSSION TITLE
How to have multiple env?

### DISCUSSION THREAD

**Discussion by jeremyhalin** [nylo-core/nylo#35] [Q&A] [answered]
I would like to have multiple `.env` files so I can have a dev version of my app plus a production version.

Is there an easy way to achieve that?

My use case is very simple, I want to be able to change the API_BASE_URL env variable in dev and production.

`String.fromEnvironment` and `--dart-define` seems to be a good option.

**Comment by agordn52** [comment, marked as answer]
An easy way would be to have separate environment files i.e `.env.developing`, `.env.production` and so on...
Then, replace the `.env` file with all the variables from the environment you want to use.

Hope that makes sense 👍

**Reply by jeremyhalin** [reply]
It makes sense, but Nylo doesn't seem to allow that. `.env` file path is hard-coded into the framework right?

**Reply by agordn52** [reply]
Create two new files at the root `.env.developing`, `.env.production` and more if you need.
Copy the env variables from the `.env` into the newly created files. After you can modify these files and then replace the contents of `.env` with whatever environment you want to use.

**Reply by jeremyhalin** [reply]
Well, of course I can do that, but I can't `flutter run` or `flutter build` one specific environment easily.
I have to replace the content of `.env` file manually?

**Reply by agordn52** [reply]
Yeah, you'll have to replace the `.env` file manually.
If you have ideas for an alternative solution, feel free to send a PR though 👍

**Reply by jeremyhalin** [reply]
I see a solution where we change signature of `init` Nylo function and add an argument like `envFile` (but env file must still be added to assets). What do you think? Would be happy to submit a PR.

But this solution still needs multiple endpoints (`main_production.dart`, `main_dev.dart`)

**Reply by jeremyhalin** [reply]
Another solution would be

1. change 
```dart
await dotenv.load(fileName: ".env");
```
by
```dart
const ENV_FILE_PATH = String.fromEnvironment('ENV_FILE_PATH', defaultValue: '.env');
await dotenv.load(fileName: ENV_FILE_PATH);
```
2. add `.env.production` to assets in `pubspec.yaml`
3. `flutter run/build --dart-define=ENV_FILE_PATH=.env.production`

**Reply by agordn52** [reply]
@jeremyhalin I think the latter solution would be sweet.
```dart
const ENV_FILE_PATH = String.fromEnvironment('ENV_FILE_PATH', defaultValue: '.env');
await dotenv.load(fileName: ENV_FILE_PATH);
```
This would work well for CI/CD pipelines too.
